solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -866,8 +876,48 @@ public BitDocSet getLiveDocSet() throws IOException {
     // but the addition of setLiveDocs means we needed to add volatile to "liveDocs".
     BitDocSet docs = liveDocs;
     if (docs == null) {
-      //note: maybe should instead calc manually by segment, using FixedBitSet.copyOf(segLiveDocs); avoid filter cache?
-      liveDocs = docs = getDocSetBits(matchAllDocsQuery);
+      switch (leafContexts.size()) {
+        case 0:
+          assert numDocs() == 0;
+          docs = new BitDocSet(BitDocSet.empty().getFixedBitSet(), 0);
```

Comment:
This is confusing about why we need to wrap a `empty()` in a new BitDocSet. Clearer if we use the super type.

```suggestion
          docs = new BitDocSet(DocSet.empty().getFixedBitSet(), 0);
```